CDTDatastore/CDTReplicator/CDTPullReplication.m

```diff
@@ -33,7 +34,16 @@ + (instancetype)replicationWithSource:(NSURL *)source target:(CDTDatastore *)tar
 - (instancetype)initWithSource:(NSURL *)source target:(CDTDatastore *)target
 {
     if (self = [super init]) {
-        _source = source;
+        
+        NSURLComponents * sourceComponents = [NSURLComponents componentsWithURL:source resolvingAgainstBaseURL:NO];
```

Comment:
You could do this in the superclass instead of in both `CDT{Pull,Push}Replication` and avoid the awkward code duplication